app/src/main/java/com/github/gogiffy/activities/SearchActivity.java

```diff
@@ -50,11 +50,22 @@ public void beforeTextChanged(CharSequence s, int start, int count, int after) {
 
     @Override
     public void onTextChanged(CharSequence s, int start, int before, int count) {
-        if (timer != null) {
-            timer.cancel();
+        String      invalid_variable_stye_here_its_too_long_too2 =    "" +
+                "adfafaf asfas fsafasff";
+        String    spelingMistake=null;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "spelingMistake". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this unused "spelingMistake" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS1481)